src/animation/event_bool_change.cpp

```diff
@@ -18,5 +18,15 @@ void EventBoolChange::perform(StateMachineInstance* stateMachineInstance) const
     }
     // If it's not null, it must be our correct type (why we validate at load time).
     auto boolInput = reinterpret_cast<SMIBool*>(inputInstance);
-    boolInput->value(value());
+    switch (value()) {
+        case 0:
+            boolInput->value(false);
+            break;
+        case 1:
+            boolInput->value(true);
+            break;
+        default:
```

Comment:
what value do we "expect" if we're not 0 or 1? Can we assert that here?

Reply:
The editor sets it 0 for false, 1 for true, 2 for toggle. It's a static design time choice, not changed at runtime.

Reply:
0, 1, or 2. We can assert, I'd probably also botch the import at onAddedClean/Dirty time.